internal/subscriber/core_test.go

```diff
@@ -0,0 +1,243 @@
+				requestCh:              requestCh,
+				ackCh:                  ackCh,
+				modAckCh:               modAckCh,
+			},
+			expected: nil,
+			wantErr:  true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockBrokerStoreCore.EXPECT().GetConsumer(gomock.Any(), gomock.Any()).
+				DoAndReturn(func(arg0 context.Context, arg1 messagebroker.ConsumerClientOptions) (messagebroker.Consumer, error) {
+					if tt.wantErr {
+						return nil, errors.New("Test Error")
+					}
+					return cs, nil
+				})
+			mockMessageBroker.EXPECT().Resume(gomock.Any(), gomock.Any()).AnyTimes().Return(nil)
+			ch.EXPECT().Get(gomock.Any(), gomock.Any()).Return([]byte{'0'}, nil).AnyTimes()
+			ch.EXPECT().Set(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+			sub, err := tt.fields.core.NewSubscriber(tt.args.ctx, tt.args.subscriberID, tt.args.subscription, tt.args.timeoutInMs, tt.args.maxOutstandingMessages, tt.args.maxOutstandingBytes, tt.args.requestCh, tt.args.ackCh, tt.args.modAckCh)
+			if err == nil {
+				assert.True(t, EqualOnly(sub, tt.expected, []string{"topic", "subscriberID", "requestChan", "ackChan", "modAckChan"}))
+			}
+			if err != nil && !tt.wantErr {
+				t.Errorf("NewSubscriber Error %v", err)
+			}
+		})
+	}
+}
+
+func TestNewCore(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockBrokerStoreCore := mocks.NewMockIBrokerStore(ctrl)
+	mockSubscriptionCore := sCore.NewMockICore(ctrl)
+	mockOffsetCore := oCore.NewMockICore(ctrl)
+	mockCache := mCache.NewMockICache(ctrl)
+
+	type args struct {
+		bs               brokerstore.IBrokerStore
+		subscriptionCore subscription.ICore
+		offsetCore       offset.ICore
+		ch               cache.ICache
+	}
+	tests := []struct {
+		name string
+		args args
+		want ICore
+	}{
+		{
+			name: "Get New subscriber Core",
+			args: args{
+				bs:               mockBrokerStoreCore,
+				subscriptionCore: mockSubscriptionCore,
+				offsetCore:       mockOffsetCore,
+				ch:               mockCache,
+			},
+			want: &Core{
+				bs:               mockBrokerStoreCore,
+				subscriptionCore: mockSubscriptionCore,
+				offsetCore:       mockOffsetCore,
+				ch:               mockCache,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, NewCore(tt.args.bs, tt.args.subscriptionCore, tt.args.offsetCore, tt.args.ch), "NewCore(%v, %v, %v, %v)", tt.args.bs, tt.args.subscriptionCore, tt.args.offsetCore, tt.args.ch)
+		})
+	}
+}
+
+func getMockConsumerCore(ctx context.Context, ctrl *gomock.Controller) *mockMB.MockConsumer {
+	cs := mockMB.NewMockConsumer(ctrl)
+	req := messagebroker.GetTopicMetadataRequest{
+		Topic:     topicName,
+		Partition: partition,
+	}
+	cs.EXPECT().GetTopicMetadata(ctx, req).Return(messagebroker.GetTopicMetadataResponse{}, nil).AnyTimes()
+	cs.EXPECT().Pause(ctx, gomock.Any()).AnyTimes().Return(nil)
+	cs.EXPECT().CommitByPartitionAndOffset(gomock.Any(), gomock.Any()).Return(
```

Comment:
try to define expect calls with pre-defined args if possible

Reply:
moved to setup